docs/usage.rst

```diff
@@ -175,6 +175,21 @@ For example::
 - ``cmake_minimum_required_version``: String identifying the minimum version of CMake required
   to configure the project.
 
+- ``cmake_target``: Target generated by CMake that will be used in the compilation step.
```

Comment:
What do you think of renaming this to `cmake_install_target` ? 

Suggestion:

```
Target generated by CMake that will be used in the compilation step.
By default, this option is set to ``install``, which is always provided by CMake.
This can be used to only install certain components.
```

->

```
Name of the target to "build" for installing the artifacts into the wheel.
By default, this option is set to ``install``, which is always provided by CMake.
This can be used to only install certain components.
```

Reply:
I like it. But not sure if you foresee any utility on using a non-install target in the future. Your call!

Reply:
Thinking a little bit more. The `install` target depends on the `build-all` internal target. So, you will never do a `make install` or `ninja install` without triggering the build-all.

However, with custom targets, like `install-runtime`, the build is not guaranteed. The user should make that new install target depends on preexisting targets to force the build. It would a nice CMake feature to expose the `build-all` target to alleviate this [issue](https://gitlab.kitware.com/cmake/cmake/-/issues/8438).

So, thinking twice, I don't think cmake_install_target is that great. Because, the user should take care of handling the dependencies of that new install target... But I am not sure, as I said, your call.

Reply:
Suggestion integrated in last amend.

Reply:
Considering that scikit-build explicitly build the install target and there is no way to add a target depending on a built-in CMake targets, the proposed feature makes sense. Thanks for referencing the CMake issue. :100:

Reply:
> the build is not guaranteed

I missed that ... 

Then in case `cmake_install_target` is specified and different from the default value,  should we do this:

```
        cmd = [self.cmake_executable, "--build", source_dir, "--config", config, "--"]
        cmd.extend(clargs)
```

followed by an other call to

```
        cmd = [self.cmake_executable, "--build", source_dir, "--target", target, "--config", config, "--"]
        cmd.extend(clargs)
```

This would ensure the project is always built before building the custom install targets.

Reply:
That is great! Integrating it...